### PR TITLE
Report cause for Akka/IO TCP CommandFailed events

### PR DESCRIPTION
**Pull request by ismaelhamed**
Port [#22954](https://github.com/akka/akka/pull/22954)

## Changes

> I think we should finally fix this once and forever. We have so many failed test jobs where we cannot be sure why "Connect failed" and we have no way of finding out without enabling DEBUG log level everywhere. That same pain is felt by every user.

## Checklist

For significant changes, please ensure that the following have been completed (delete if not relevant):

* [X] This change follows the [Akka.NET API Compatibility Guidelines](https://getakka.net/community/contributing/api-changes-compatibility.html).
* [X] I have [reviewed my own pull request](https://getakka.net/community/contributing/index.html#review-your-own-pull-requests).
* [X] Changes in public API reviewed, if any.

### CODE REVIEW

File: src/core/Akka/IO/Tcp.cs [hunk 1]
```diff
@@ -808,18 +810,32 @@ public sealed class CommandFailed : Event
...
+            public Option<Exception> Cause { get; private set; } = Option<Exception>.None;
+
+            /// <summary>
+            /// Creates a copy of this object with a new cause set.
+            /// </summary>
+            [InternalApi]
+            public CommandFailed WithCause(Exception cause)
+            {
+                // Needs to be added with a mutable property for compatibility reasons
+                return new CommandFailed(Cmd) { Cause = cause };
```
Comment (ismaelhamed) [inline src/core/Akka/IO/Tcp.cs:832]: Not sure if this holds true in C# or we could just add a second constructor instead.

Comment (Aaronontheweb) [reply src/core/Akka/IO/Tcp.cs:832]: We could just add a constructor overload and that would work

Comment (Aaronontheweb) [reply src/core/Akka/IO/Tcp.cs:832]: Although I think this is probably a fine way of doing it

**Review by Aaronontheweb — APPROVED** [review]
LGTM

**Comment by Aaronontheweb** [comment]
@Arkatufus can we backport to v1.4?

**Pull request merged**